### PR TITLE
Merge in Sockeye Autopilot

### PR DESCRIPTION
**Pull request by mjdenkowski**
This merges in Sockeye Autopilot, a module that provides single-command end-to-end NMT system building.  See [`contrib/autopilot/README.md`](https://github.com/awslabs/sockeye/tree/autopilot/contrib/autopilot/README.md) for a full description.

This is a `contrib` module that depends on but does not change the `sockeye` module.  Required third-party tools are downloaded at runtime, removing the need to check these tools into the Sockeye repository.  Autopilot passes the same checks as the core code and has its own system tests, `contrib/test.py`.  Any feedback is welcome, though this probably doesn't require the same level of detailed review as core Sockeye code.

#### Pull Request Checklist ##
- [x] Changes are complete (if posting work-in-progress code, prefix your pull request title with '[WIP]'
until you can check this box.
- [x] Unit tests pass (`pytest`)
- [x] System tests pass (`pytest test/system`)
- [x] Passed code style checking (`./style-check.sh`)
- [x] `contrib/autopilot/test.py` passes
- [x] Updated major/minor version in `sockeye/__init__.py`. Major version bump if this is a backwards incompatible change.
- [x] Updated CHANGELOG.md

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,12 @@ Note that Sockeye has checks in place to not translate with an old model that wa
 
 Each version section may have have subsections for: _Added_, _Changed_, _Removed_, _Deprecated_, and _Fixed_.
 
+## [1.18.14]
+
```
Comment (fhieber) [inline CHANGELOG.md]: could you remove those newlines? I accidentally also added one below before.

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,12 @@ Note that Sockeye has checks in place to not translate with an old model that wa
...
+### Added
+- Introduced Sockeye Autopilot for single-command end-to-end system building.
+See the [Autopilot documentation]((https://github.com/awslabs/sockeye/tree/master/contrib/autopilot)) and run with: `sockeye-autopilot`.
+
 ## [1.18.13]
```
Comment (fhieber) [inline CHANGELOG.md]: same here

File: CHANGELOG.md [hunk 3]
```diff
@@ -10,6 +10,12 @@ Note that Sockeye has checks in place to not translate with an old model that wa
 
 Each version section may have have subsections for: _Added_, _Changed_, _Removed_, _Deprecated_, and _Fixed_.
 
+## [1.18.14]
+
+### Added
+- Introduced Sockeye Autopilot for single-command end-to-end system building.
```
Comment (fhieber) [inline CHANGELOG.md]: Can we mention that this module was added to `contrib`, meaning that it is not tested with every PR?

File: contrib/autopilot/autopilot.py [hunk 4]
```diff
@@ -0,0 +1,902 @@
...
+def renew_step_dir(step_dir: str):
+    """Delete step directory if exists and create, reporting actions."""
+    if os.path.exists(step_dir):
+        logging.info("Remove unfinished step %s", step_dir)
+        shutil.rmtree(step_dir)
+    logging.info("Create: %s", step_dir)
+    os.makedirs(step_dir)
+
+
+def call_sockeye_train(args: List[str],
```
Comment (fhieber) [inline contrib/autopilot/autopilot.py]: Not relevant for this PR, but this task may benefit from the recently added feature of using config files and being able to start training as an API call (`sockeye.train.train()` with parsed args directly).

**Comment by fhieber** [comment]
How long does a full run of `autopilot` take in TestMode and how much disk space does it need? Should we consider running this as a nightly/weekly Travis cron job if fast enough?

File: CHANGELOG.md (hunk 3 above)
Comment (mjdenkowski) [reply CHANGELOG.md]: Sure, we can be clear about that in the documentation.

File: contrib/autopilot/autopilot.py (hunk 4 above)
Comment (mjdenkowski) [reply contrib/autopilot/autopilot.py]: That's a very good idea.  We can follow up in another PR.

**Comment by mjdenkowski** [comment]
Good call--we can definitely trim down a test task to run nightly.  That and config file integration can roll into a second PR.

**Comment by tdomhan** [comment]
How about actually either linking to the README from `/tutorials` or add a tutorial in order to promote this feature?

**Pull request merged**